test/integration/scheduler/serving/endpoints_test.go

```diff
@@ -175,29 +176,44 @@ func TestEndpointHandlers(t *testing.T) {
 					req.Header.Set(k, v)
 				}
 			}
-			r, err := client.Do(req)
-			if err != nil {
-				t.Fatalf("failed to GET %s from component: %v", tt.path, err)
-			}
 
-			body, err := io.ReadAll(r.Body)
+			err = retry.OnError(
```

Comment:
it feels a better fix to expect that the process under test is ready after calling `kubeschedulertesting.StartTestServer(` and not have to retry in the test


https://github.com/kubernetes/kubernetes/blob/30469e180361d7da07b0fee6d47c776fa2cf3e86/cmd/kube-scheduler/app/testing/testserver.go#L159-L164

Reply:
@aojea but the test case `/readyz with broken apiserver` expect the readyz endpoint to return 500 when it is started with a broken apiserver. If I add a check for readyz in the function, we have to change the func signature to accept a new parameter which is used to control whether to check for readyz or not.

Reply:
but the test case `/readyz with broken apiserver` expect the readyz endpoint to return 500 when it is started with a broken apiserver. If I add a check for readyz in the function, we have to change the func signature to accept a new parameter which is used to control whether to check for readyz or not.

Reply:
I see, sorry, didn;t read the test in detail , so with this change, do we only retry when the config is not broken and the error is `handlers are not fully synchronized`?

is the case we only need to retry on ready endpoints only? if affirmative, we should make that part of the retry function

Reply:
updated.